tools/infrastructure/api_compare.py

```diff
@@ -0,0 +1,368 @@
+        else:
+            filter_func = filter[param]
+            if filter_func(rpc):
+                filtered.append(param)
+    return filtered
+
+
+def CreateResultSection(param, result):
+    if param not in result:
+        result[param] = {}
+    if "HMI_API" not in result[param]:
+        result[param]["HMI_API"] = {}
+    if "Mobile_API" not in result[param]:
+        result[param]["Mobile_API"] = {}
+    mob_api_result_section = result[param]["Mobile_API"]
+    hmi_api_result_section = result[param]["HMI_API"]
+    return mob_api_result_section, hmi_api_result_section
+
+
+def CompareParams(param, mob_param_attrs, hmi_param_attrs):
+    result = {}
+    mob_param_attrs = dict(mob_param_attrs)
+    hmi_param_attrs = dict(hmi_param_attrs)
+    for compare_rule in global_compare_rules:
+        condition = compare_rule[0]
+        rule = compare_rule[1]
+        if (condition(mob_param_attrs, hmi_param_attrs)):
+            mobile_res, hmi_res = rule(
+                mob_param_attrs, hmi_param_attrs)
+            if len(mobile_res) > 0 or len(hmi_res) > 0:
+                m_res, h_res = CreateResultSection(param, result)
+                m_res.update(mobile_res)
+                h_res.update(hmi_res)
+    return result
+
+
+def FillSummaryResult(rpc_name, area, value):
+    if rpc_name not in summary_result:
+        summary_result[rpc_name] = {}
+    rpc = summary_result[rpc_name]
+    if type(value) is not dict:
+        rpc[area] = value
+    elif type(value) is dict:
+        if area not in rpc:
+            rpc[area] = {}
+        rpc[area].update(value)
+
+
+ExtractFunctions(mobile_root, mob_apis)
+
+for interface in hmi_root:  # loop by every interface
+    ExtractFunctions(interface, hmi_apis)
+
+
+for func_name in sorted(mob_apis.keys()):
+    if func_name in hmi_apis:
+        hmi_rpc_params = hmi_apis[func_name]
+        mob_rpc_params = mob_apis[func_name]
+        rpc = type('', (), {})()  # empty object
+        rpc.mobile_params = mob_rpc_params
+        rpc.hmi_rpc_params = hmi_rpc_params
+        rpc.rpc_name = func_name.rpc_name
+        rpc.rpc_type = func_name.rpc_type
+        mob_keys = mob_rpc_params.keys()
+        hmi_keys = hmi_rpc_params.keys()
+        common_parameters_list = list(
+            set(mob_keys).intersection(set(hmi_keys)))
+        absent_params_mob_api = list(
+            set(hmi_rpc_params.keys()) - set(mob_rpc_params.keys()))
+        absent_params_mob_api = FilterParams(
+            rpc, absent_params_mob_api, exclude_params_from_mob_rpc)
+        absent_params_hmi_api = set(
+            mob_rpc_params.keys()) - set(hmi_rpc_params.keys())
+        absent_params_hmi_api = FilterParams(
+            rpc, absent_params_hmi_api, exclude_redundant_from_hmi_rpc_filter)
+        print "\n" + "---" * 60 + "\n"
+        rpc_color = colors.BOLD + colors.HEADER
+        func_name = str(func_name.rpc_name + " " + func_name.rpc_type)
+        print rpc_color + func_name + colors.ENDC
```

Comment:
@AByzhynar Please try first to do all logic and then print all debug information